addons/survey/controllers/main.py

```diff
@@ -228,45 +212,46 @@ def get_scores(self, survey, token, page=None, **post):
     def submit(self, survey, **post):
         _logger.debug('Incoming data: %s', post)
         page_id = int(post['page_id'])
-        cr, uid, context = request.cr, request.uid, request.context
-        survey_obj = request.registry['survey.survey']
-        questions_obj = request.registry['survey.question']
-        questions_ids = questions_obj.search(cr, uid, [('page_id', '=', page_id)], context=context)
-        questions = questions_obj.browse(cr, uid, questions_ids, context=context)
+        Survey = request.env['survey.survey']
+        Question = request.env['survey.question']
+        questions = Question.search([('page_id', '=', page_id)])
 
         # Answer validation
         errors = {}
         for question in questions:
             answer_tag = "%s_%s_%s" % (survey.id, page_id, question.id)
-            errors.update(questions_obj.validate_question(cr, uid, question, post, answer_tag, context=context))
+            errors.update(Question.validate_question(question, post, answer_tag))
```

Comment:
Why not call it with question, as question is nothing but recordset of survey.question.
``` question.validate_question(post, answer_tag) ```
and signature of validate_question will be
``` def validate_question(self, post, answer_tag): ```